diff --git a/src/generic/serial_irq.c b/src/generic/serial_irq.c
--- a/src/generic/serial_irq.c
+++ b/src/generic/serial_irq.c
@@ -17,7 +17,7 @@
 #define RX_BUFFER_SIZE 192
 
 static uint8_t receive_buf[RX_BUFFER_SIZE], receive_pos;
-static uint8_t transmit_buf[96], transmit_pos, transmit_max;
+static uint8_t transmit_buf[192], transmit_pos, transmit_max;
 
 DECL_CONSTANT("SERIAL_BAUD", CONFIG_SERIAL_BAUD);
 DECL_CONSTANT("RECEIVE_WINDOW", RX_BUFFER_SIZE);
