diff --git a/klippy/extras/homing.py b/klippy/extras/homing.py
--- a/klippy/extras/homing.py
+++ b/klippy/extras/homing.py
@@ -424,12 +424,13 @@ class PrinterHoming:
         gcode.register_command("G28", self.cmd_G28)
 
     def manual_home(
-        self, toolhead, endstops, pos, speed, triggered, check_triggered
+        self, toolhead, endstops, pos, speed, probe_pos, triggered, check_triggered
     ):
         hmove = HomingMove(self.printer, endstops, toolhead)
         try:
-            hmove.homing_move(
-                pos, speed, triggered=triggered, check_triggered=check_triggered
+            epos = hmove.homing_move(
+                pos, speed, probe_pos=probe_pos, triggered=triggered,
+                check_triggered=check_triggered
             )
         except self.printer.command_error:
             if self.printer.is_shutdown():
@@ -437,6 +438,7 @@ class PrinterHoming:
                     "Homing failed due to printer shutdown"
                 )
             raise
+        return epos
 
     def probing_move(self, mcu_probe, pos, speed):
         endstops = [(mcu_probe, "probe")]
