diff --git a/klippy/extras/danger_options.py b/klippy/extras/danger_options.py
--- a/klippy/extras/danger_options.py
+++ b/klippy/extras/danger_options.py
@@ -33,7 +33,7 @@ class DangerOptions:
             "single_mcu_trsync_timeout", 0.25, minval=0.0
         )
         self.multi_mcu_trsync_timeout = config.getfloat(
-            "multi_mcu_trsync_timeout", 0.025, minval=0.0
+            "multi_mcu_trsync_timeout", 0.050, minval=0.0
         )
         self.homing_elapsed_distance_tolerance = config.getfloat(
             "homing_elapsed_distance_tolerance", 0.5, minval=0.0
