diff --git a/scripts/buildcommands.py b/scripts/buildcommands.py
index 3c5528c..eaa1d28 100644
--- a/scripts/buildcommands.py
+++ b/scripts/buildcommands.py
@@ -594,6 +594,9 @@ def git_version():
 
 
 def build_version(extra, cleanbuild):
+    version = os.environ.get("KALICO_BUILD_VERSION")
+    if version:
+        return version + extra
     version = git_version()
     if not version:
         cleanbuild = False
